### PR TITLE
feat: use Processor for data persister

### PR DESCRIPTION
**Pull request by samuelbigard**
| Q             | A
| ------------- | ---
| Branch?       | main 
| Type       | Feature
| Tickets       | 
| License       | MIT

### CODE REVIEW

File: src/Core/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml [hunk 1]
```diff
@@ -15,6 +15,11 @@
             <tag name="api_platform.data_persister" priority="-1000" />
         </service>
```
Comment (dunglas) [inline src/Core/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml:16]: this service must be deprecated

File: src/Core/Bridge/Symfony/Bundle/Resources/config/state.xml [hunk 2]
```diff
@@ -20,8 +20,9 @@
         </service>
 
         <service id="api_platform.state.legacy_data_persister_processor" class="ApiPlatform\State\LegacyDataPersisterProcessor">
```
Comment (soyuka) [inline src/Core/Bridge/Symfony/Bundle/Resources/config/state.xml:22]: a supprimer

File: src/Core/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig [hunk 3]
```diff
@@ -194,9 +194,9 @@
                 </div>
             </div>
             <div class="tab">
-                <h3 class="tab-title data-persister-tab-title">Data Persisters</h3>
-                <div class="tab-content data-persister-tab-content">
-                    {{ apiPlatform.providerTable(collector.dataPersisters, 'data persister') }}
+                <h3 class="tab-title data-processor-tab-title">Processors</h3>
+                <div class="tab-content data-processor-tab-content">
+                    {{ apiPlatform.providerTable(collector.processors, 'processors') }}
                 </div>
```
Comment (soyuka) [inline src/Core/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig:200]: keep data persister

File: src/Symfony/Messenger/Processor.php [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    private $resourceMetadataFactory;
+
+    public function __construct($resourceMetadataFactory, MessageBusInterface $messageBus)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->messageBus = $messageBus;
+
+        if (!$resourceMetadataFactory instanceof ResourceMetadataCollectionFactoryInterface) {
+            trigger_deprecation('api-platform/core', '2.7', sprintf('Use "%s" instead of "%s".', ResourceMetadataCollectionFactoryInterface::class, ResourceMetadataFactoryInterface::class));
+        }
```
Comment (soyuka) [inline src/Symfony/Messenger/Processor.php:47]: remove deprec

File: src/Symfony/Messenger/Processor.php [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+{
+    use ClassInfoTrait;
+    use DispatchTrait;
+
+    /**
+     * @var ResourceMetadataCollectionFactoryInterface|ResourceMetadataFactoryInterface
+     */
+    private $resourceMetadataFactory;
+
+    public function __construct($resourceMetadataFactory, MessageBusInterface $messageBus)
```
Comment (soyuka) [inline src/Symfony/Messenger/Processor.php:40]:
```suggestion
    public function __construct(ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory, MessageBusInterface $messageBus)
```

File: src/Symfony/Messenger/Processor.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        if ($this->resourceMetadataFactory instanceof ResourceMetadataCollectionFactoryInterface) {
+            try {
+                $resourceMetadataCollection = $this->resourceMetadataFactory->create($context['resource_class'] ?? $this->getObjectClass($data));
+                $operation = isset($context['graphql_operation_name']) ? $resourceMetadataCollection->getGraphQlOperation($context['graphql_operation_name']) : $resourceMetadataCollection->getOperation($operationName ?? null);
+
+                return false !== ($operation->getMessenger() ?? false);
+            } catch (OperationNotFoundException $e) {
+                return false;
+            }
```
Comment (soyuka) [inline src/Symfony/Messenger/Processor.php:103]: keep only this code

File: tests/Fixtures/TestBundle/Document/ProcessorEntity.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Fixtures\TestBundle\Document;
+
+use ApiPlatform\Core\Annotation\ApiResource;
```
Comment (soyuka) [inline tests/Fixtures/TestBundle/Document/ProcessorEntity.php:16]:
```suggestion
use ApiPlatform\Metadata\ApiResource;
```

**Review by soyuka — COMMENTED** [review]
:+1:

**Pull request merged**